src/ConcurrentLinkedList/ConcurrentLinkedList.cs

```diff
@@ -122,15 +122,16 @@ private static bool HelpRemove(Node<T> node, T value, out T result)
                     previous.Next = successor;
                     current = successor;
                 }
+                else if (state == NodeState.REM)
```

Comment:
I don't think moving this check here is right. It solves the immediate problem, but stops threads finding the nodes they are looking to remove (hence the failing tests). I think solution is to change the node value check to also check for null, as you did in first commit. 

So `else if (!current.Value.Equals(value))` => `else if (current.Value != null && !current.Value.Equals(value))` and move this check back to where it was. Your added test passes with this as well.